File: modules/spring-core/testdata/java/inspection/autowired/src/qualifier/TestAutowired.java
package component_scan.qualifier;

import org.springframework.stereotype.Component;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.beans.factory.annotation.Qualifier;

@Configuration
class MyComponent {
    @Qualifier("fooBean")
    @Autowired
    FooInterface testQualifier1;
    
    @Qualifier("not")
    @Autowired
    FooInterface testQualifier2;
    
    @Qualifier("nameQualifier")
    @Autowired
    FooInterface testQualifier3;
}

interface FooInterface {
}

@Component
class FooBean implements FooInterface {
}

@Component
class OtherBean implements FooInterface {
}

@Component
@Qualifier("nameQualifier")
class AnotherBean implements FooInterface {
}

File: modules/spring-core/testdata/kotlin/inspection/autowired/src/component_scan/qualifier/TestAutowired.kt
package component_scan.qualifier

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.annotation.Configuration
import org.springframework.stereotype.Component

@Configuration
open class MyComponent {
    @Qualifier("fooBean")
    @Autowired
    var testQualifier1: FooInterface? = null

    @Qualifier("not")
    @Autowired
    var testQualifier2: FooInterface? = null

    @Qualifier("nameQualifier")
    @Autowired
    var testQualifier3: FooInterface? = null
}

interface FooInterface

@Component
open class FooBean : FooInterface

@Component
open class OtherBean : FooInterface

@Component
@Qualifier("nameQualifier")
open class AnotherBean : FooInterface